File: RouteZoekerC/Pad.cs
using System.Drawing;

class Pad
{
    public Stad Hier;
    public Pad Rest;
    public int Kosten;

    public Pad(Stad hier, Pad rest, int k)
    {   Hier = hier;
        Rest = rest;
        Kosten = k;
        if (rest != null) 
            Kosten += rest.Kosten;
    }
    public bool Bevat(Stad s)
    {   if (Hier == s) return true;
        if (Rest == null) return false;
        return Rest.Bevat(s);
    }
    public void Teken(Graphics gr, Brush brush, Pen pen)
    {   Hier.Teken(gr,brush);
        if (Rest != null)
        {   gr.DrawLine(pen, Hier.Plek, Rest.Hier.Plek);
            Rest.Teken(gr, brush, pen);
        }
    }
}

File: RouteZoekerP/Pad.py
class Pad:
    def __init__(self, hier, rest, kosten):
        self.Hier = hier
        self.Rest = rest
        self.Kosten = kosten
        if rest:
            self.Kosten += rest.Kosten

    def Bevat(self, stad):
        if self.Hier==stad:
            return True
        if self.Rest==None:
            return False
        return self.Rest.Bevat(stad)

    def Teken(self, draw, kleur):
        self.Hier.Teken(draw, kleur)
        if self.Rest:
            draw.line((self.Hier.Plek, self.Rest.Hier.Plek), kleur)
            self.Rest.Teken(draw, kleur)
